src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -619,8 +622,7 @@ private[formal] object RepeatingInterval {
     case IsoFields.QUARTER_YEARS =>
       ldt.withMonth((ldt.getMonthValue - 1) / 4 + 1).withDayOfMonth(1).truncatedTo(ChronoUnit.DAYS)
     case ChronoUnit.MONTHS => ldt.withDayOfMonth(1).truncatedTo(ChronoUnit.DAYS)
-    case ChronoUnit.WEEKS =>
-      ldt.withDayOfYear(ldt.getDayOfYear - ldt.getDayOfWeek.getValue + 1).truncatedTo(ChronoUnit.DAYS)
+    case ChronoUnit.WEEKS => ldt.withDayOfYear((ldt.getDayOfYear - ldt.getDayOfWeek.getValue + 1) max 1).truncatedTo(ChronoUnit.DAYS)
```

Comment:
Please add a test for this. Make sure, for example, that the week of Tuesday January 1st is truncated to Monday December 31st of the preceding year.

Reply:
I have added a test and it seems to work well.